network/dnscrypt-proxy/dnscrypt-proxy.SlackBuild

```diff
@@ -22,101 +23,93 @@
-  echo "$PRGNAM-$VERSION-$ARCH-$BUILD$TAG.$PKGTYPE"
-  exit 0
+    echo "$PRGNAM-$VERSION-$ARCH-$BUILD$TAG.$PKGTYPE"
+    exit 0
 fi
 
 TMP=${TMP:-/tmp/SBo}
 PKG=$TMP/package-$PRGNAM
 OUTPUT=${OUTPUT:-/tmp}
 
-if [ "$ARCH" = "i586" ]; then
-  LIBDIRSUFFIX=""
-elif [ "$ARCH" = "i686" ]; then
-  LIBDIRSUFFIX=""
-elif [ "$ARCH" = "x86_64" ]; then
-  LIBDIRSUFFIX="64"
-else
-  LIBDIRSUFFIX=""
+bailout() {
+    echo -e "\nYou must have a $DNSCRYPT_USER user and $DNSCRYPT_GROUP group to run this script. "
+    echo -e    "Something like this should suffice for most systems: "
+    echo -e    "# groupadd -g $DNSCRYPT_GID $DNSCRYPT_GROUP "
+    echo -e    "# useradd -u $DNSCRYPT_UID -g $DNSCRYPT_GID -c \"DNSCrypt-proxy\" -d /dev/null -s /bin/false $DNSCRYPT_USER \n"
+    exit 1
+}
+
+if ! grep -q "^$DNSCRYPT_USER:" /etc/passwd; then
+    bailout
+elif ! grep -q "^$DNSCRYPT_GROUP:" /etc/group; then
+    bailout
 fi
 
-set -e
-
 rm -rf $PKG
-mkdir -p $TMP $PKG $OUTPUT
+mkdir -p $TMP/$PRGNAM-$VERSION $PKG $OUTPUT
 cd $TMP
 rm -rf $PRGNAM-$VERSION
-mkdir -p $PRGNAM-$VERSION/src/$DOMAIN/$ORG
-cd $PRGNAM-$VERSION/src/$DOMAIN/$ORG
-tar xvf $CWD/$PRGNAM-$VERSION.tar.gz
-mv $PRGNAM-$VERSION $PRGNAM
-cd $TMP/$PRGNAM-$VERSION
+
+if [[ $ARCH == i?86 ]]; then
+    tar xvf $CWD/$PRGNAM-linux_i386-$VERSION.tar.gz --transform="s/linux-i386/$PRGNAM-$VERSION/"
+elif [[ $ARCH == x86_64 ]]; then
+    tar xvf $CWD/$PRGNAM-linux_x86_64-$VERSION.tar.gz --transform="s/linux-x86_64/$PRGNAM-$VERSION/"
+else
+    echo "Architecture not supported"
+    exit 1
+fi
+
+cd $PRGNAM-$VERSION
 chown -R root:root .
+
 find -L . \
  \( -perm 777 -o -perm 775 -o -perm 750 -o -perm 711 -o -perm 555 \
   -o -perm 511 \) -exec chmod 755 {} \; -o \
  \( -perm 666 -o -perm 664 -o -perm 640 -o -perm 600 -o -perm 444 \
   -o -perm 440 -o -perm 400 \) -exec chmod 644 {} \;
 
-export GO111MODULE=auto
-export GOPATH="$TMP/$PRGNAM-$VERSION"
-
-go install -a -x ./...
-
-cd $TMP/$PRGNAM-$VERSION/src/$DOMAIN/$ORG/$PRGNAM
-
-mkdir -p $PKG/usr/sbin
+mkdir -p $PKG/usr/{bin,doc/$PRGNAM-$VERSION} \
+    $PKG/etc/{$PRGNAM,rc.d} \
+    $PKG/var/{log,run}/$PRGNAM \
+    $PKG/install
 
-install -m 755 "$TMP/$PRGNAM-$VERSION/bin/${PRGNAM}" $PKG/usr/sbin/${PRGNAM}
+cp -a *.txt $PKG/etc/$PRGNAM/
+install -D -m 0644 LICENSE $PKG/usr/doc/$PRGNAM-$VERSION/LICENSE
+install -D -m 0755 $PRGNAM $PKG/usr/bin/$PRGNAM
+install -D -m 0644 $CWD/$PRGNAM.toml $PKG/etc/$PRGNAM/$PRGNAM.toml
+install -D -m 0644 $CWD/README.Slackware $PKG/usr/doc/$PRGNAM-$VERSION/README.Slackware
+install -D -m 0644 $CWD/$PRGNAM.SlackBuild $PKG/usr/doc/$PRGNAM-$VERSION/$PRGNAM.SlackBuild
+install -D -m 0644 $CWD/rc.$PRGNAM $PKG/etc/rc.d/rc.$PRGNAM
 
-find $PKG -print0 | xargs -0 file | grep -e "executable" -e "shared object" | grep ELF \
-  | cut -f 1 -d : | xargs strip --strip-unneeded 2> /dev/null || true
+sed -i "s/DNSCRYPT_USER=.*$/DNSCRYPT_USER=$(sed -n 's/DNSCRYPT_USER=\${DNSCRYPT_USER:-\([^}]*\)}/\1/p' $CWD/$PRGNAM.SlackBuild)/" $CWD/doinst.sh
```

Comment:
Just use the variable you already have assigned:

```
sed -i "s/DNSCRYPT_USER=.*$/DNSCRYPT_USER=$DNSCRYPT_USER/" $CWD/doinst.sh
```

But I think also you should switch these to use the ids instead of names.
$DNSCRYPT_GID and $DNSCRYPT_UID